app/controllers/nsm/imports_controller.rb

```diff
@@ -71,10 +102,29 @@ def xml_file
       @xml_file ||= Nokogiri::XML::Document.parse(@form_object.file_upload.tempfile.read, &:noblanks)
     end
 
+    # Extract version from XML
+    def extract_version_from_xml
+      version_node = xml_file.at_xpath('//version')
+      return version_node.text.to_i if version_node&.text.present?
+
+      1 # Default to version 1 if not found
```

Comment:
not feeling comfortable with the fact that we are defaulting to version 1 if not found.  We should always be getting the version with the xml file, there is even logic in the validate method that adds an error if the version node is empty - meaning it shouldn't be possible to get to this point if this is the case (therefore, why add code for it?) - this could cause us to slip up when making future updates - if this is for testing purposes, I feel like there's a better way

Reply:
Forgot to delete this, already added all the case scenarios check like , missing, invalid alongside valid and unsupported.